ts/components/session/SessionIcon.tsx

```diff
@@ -0,0 +1,241 @@
+    viewBox: '0 0 22 21',
+  },
+};
+
+export enum SessionIconSizes {
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+interface Props {
+  iconType: SessionIconTypes;
+  iconSize: SessionIconSizes;
+  iconColor: string;
+  iconRotation: number;
+  onClick: any;
+}
+
+export class SessionIcon extends React.PureComponent<Props> {
+  public static defaultProps = {
+    iconSize: SessionIconSizes.Medium,
+    iconColor: '',
+    iconRotation: 0,
+  };
+
+  constructor(props: any) {
+    super(props);
+  }
+
+  public render() {
+    const { iconType, iconSize, iconColor, iconRotation } = this.props;
+
+    let iconDimensions;
+    switch (iconSize) {
+      case SessionIconSizes.Small:
+        iconDimensions = '15';
+        break;
+      case SessionIconSizes.Medium:
+        iconDimensions = '20';
+        break;
+      case SessionIconSizes.Large:
+        iconDimensions = '30';
+        break;
+      default:
+        iconDimensions = '20';
+    }
+
+    let iconDef = { path: '', viewBox: '' };
+
+    switch (iconType) {
+      case SessionIconTypes.Arrow:
+        iconDef = icons.Arrow;
+        break;
+      case SessionIconTypes.ChatBubble:
+        iconDef = icons.ChatBubble;
+        break;
+      case SessionIconTypes.Check:
+        iconDef = icons.Check;
+        break;
+      case SessionIconTypes.Chevron:
+        iconDef = icons.Chevron;
+        break;
+      case SessionIconTypes.CirclePlus:
+        iconDef = icons.CirclePlus;
+        break;
+      case SessionIconTypes.Contacts:
+        iconDef = icons.Contacts;
+        break;
+      case SessionIconTypes.Ellipses:
+        iconDef = icons.Ellipses;
+        break;
+      case SessionIconTypes.Emoji:
+        iconDef = icons.Emoji;
+        break;
+      case SessionIconTypes.Eye:
+        iconDef = icons.Eye;
+        break;
+      case SessionIconTypes.File:
+        iconDef = icons.File;
+        break;
+      case SessionIconTypes.Gear:
+        iconDef = icons.Gear;
+        break;
+      case SessionIconTypes.Globe:
+        iconDef = icons.Globe;
+        break;
+      case SessionIconTypes.Microphone:
+        iconDef = icons.Microphone;
+        break;
+      case SessionIconTypes.Moon:
+        iconDef = icons.Moon;
+        break;
+      case SessionIconTypes.Reply:
+        iconDef = icons.Reply;
+        break;
+      case SessionIconTypes.Search:
+        iconDef = icons.Search;
+        break;
+      case SessionIconTypes.Send:
+        iconDef = icons.Send;
+        break;
+      case SessionIconTypes.Star:
+        iconDef = icons.Star;
+        break;
+      default:
+        return;
+    }
+
+    const styles = {
+      transform: `rotate(${iconRotation}deg)`,
+    };
+
+    return (
+      <svg
+        className={classNames('session-icon', iconType, iconRotation)}
```

Comment:
Do we actually need to pass `iconRotation` as a class?

Reply:
Good point. Amended.